rust-install/src/component/components.rs

```diff
@@ -36,22 +40,23 @@ impl<'a> ChangeSet<'a> {
 	}
 }
 
-#[derive(Clone)]
+#[derive(Clone, Debug)]
 pub struct Components {
 	prefix: InstallPrefix,
 }
 
 impl Components {
-	pub fn new(prefix: InstallPrefix) -> Option<Self> {
-		if utils::is_file(prefix.manifest_file(COMPONENTS_FILE)) {
-			Some(Components { prefix: prefix })
-		} else {
-			None
-		}
-	}
-	pub fn init(prefix: InstallPrefix) -> Result<Self> {
-		try!(utils::write_file("components", &prefix.manifest_file(COMPONENTS_FILE), ""));
-		Ok(Components { prefix: prefix })
```

Comment:
I didn't change Configuration and manifestation because I haven't started learning that code yet, but I can also change them today.